pyuvdata/uvcal/initializers.py

```diff
@@ -25,20 +24,21 @@
 
 def new_uvcal(
     *,
-    antenna_positions: np.ndarray | dict[str | int, np.ndarray],
-    telescope_location: Locations,
-    telescope_name: str,
     cal_style: Literal["sky", "redundant"],
     gain_convention: Literal["divide", "multiply"],
     jones_array: np.ndarray | str,
-    x_orientation: Literal["east", "north", "e", "n", "ew", "ns"],
     time_array: np.ndarray | None = None,
     time_range: np.ndarray | None = None,
     freq_array: np.ndarray | None = None,
     freq_range: np.ndarray | None = None,
     cal_type: Literal["delay", "gain"] | None = None,
     integration_time: float | np.ndarray | None = None,
     channel_width: float | np.ndarray | None = None,
+    telescope: Telescope | None = None,
+    telescope_location: Locations | None = None,
+    telescope_name: str | None = None,
+    x_orientation: Literal["east", "north", "e", "n", "ew", "ns"] | None = None,
+    antenna_positions: np.ndarray | dict[str | int, np.ndarray] | None = None,
```

Comment:
Here I would argue that we should deprecate the passing of the `telescope_` params... let the user just pass a `Telescope` object (which should be able to be created as a one-liner with `Telescope.from_known_telescopes()` or some other simple method whether you pass the parameters. This reduces redundant logic. You probably _could_ accept the `telescope` parameter as a string, which would be expected to match a known telescope name.

Reply:
Obviously we can't do that without jumping a major version, but you could raise warnings from now.

Reply:
My point is it's not harder for the user to call this:

```python
uvc = UVCal.new(
    telescope = Telescope.from_params(
        location=...,
        name='blah',
        ...
    ),
    ...
)
```

than this:

```python
uvc = UVCal.new(
    telescope_location=...,
    telescope_name=....,
    ...
)
```

But, the former reduces code logic significantly, and also subtly informs the user that what's really going on under the hood is that there's a Telescope object, which is conceptually simpler for them too.

Reply:
They can just pass the telescope object and if they do, all the other things are just ignored. I left them there to avoid breaking code but I can deprecate the other options if you like.

Reply:
They are now deprecated.